doc/guides/maintaining-npm.md

```diff
@@ -0,0 +1,50 @@
+# Maintaining npm in Node.js
+
+## Step 1: Clone npm
+
+    $ git clone https://github.com/npm/npm.git
+    $ cd npm
+    
+or if you already have npm cloned make sure the repo is up to date
+
+    $ git remote update -p
+    $ git reset --hard origin master
+
+## Step 2: Build release
+
+    $ git checkout vX.Y.Z
+    $ make release
+
+## Step 3: Rimraf old npm
+    
+    $ cd ../node
```

Comment:
Nit: I would add a comment explaining that the assumption is that the node repo is in the same directory of the npm repo.

Reply:
Or just `cd /path/to/node`